packages/flutter/test/gestures/gesture_binding_resample_event_on_widget_test.dart

```diff
@@ -0,0 +1,118 @@
+    Duration now() => Duration(milliseconds: binding.clock.now().millisecondsSinceEpoch);
+    final Duration epoch = now();
+    final ui.PointerDataPacket packet = ui.PointerDataPacket(
+      data: <ui.PointerData>[
+        ui.PointerData(
+            change: ui.PointerChange.add,
+            physicalX: 0.0,
+            timeStamp: epoch + const Duration(milliseconds: 0),
+        ),
+        ui.PointerData(
+            change: ui.PointerChange.down,
+            physicalX: 0.0,
+            timeStamp: epoch + const Duration(milliseconds: 1),
+        ),
+        ui.PointerData(
+            change: ui.PointerChange.move,
+            physicalX: 10.0,
+            timeStamp: epoch + const Duration(milliseconds: 2),
+        ),
+        ui.PointerData(
+            change: ui.PointerChange.move,
+            physicalX: 20.0,
+            timeStamp: epoch + const Duration(milliseconds: 3),
+        ),
+        ui.PointerData(
+            change: ui.PointerChange.move,
+            physicalX: 30.0,
+            timeStamp: epoch + const Duration(milliseconds: 4),
+        ),
+        ui.PointerData(
+            change: ui.PointerChange.up,
+            physicalX: 40.0,
+            timeStamp: epoch + const Duration(milliseconds: 5),
+        ),
+        ui.PointerData(
+            change: ui.PointerChange.remove,
+            physicalX: 40.0,
+            timeStamp: epoch + const Duration(milliseconds: 6),
+        ),
+      ],
+    );
+
+    final List<PointerEvent> events = <PointerEvent>[];
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: Listener(
+          onPointerDown: (PointerDownEvent event) => events.add(event),
+          onPointerMove: (PointerMoveEvent event) => events.add(event),
+          onPointerUp: (PointerUpEvent event) => events.add(event),
+          child: const Text('test'),
+        ),
+      ),
+    );
+
+    GestureBinding.instance.resamplingEnabled = true;
+    GestureBinding.instance.samplingOffset = const Duration(microseconds: -5500);
+    ui.window.onPointerDataPacket(packet);
+    expect(events.length, 0);
+
+    await tester.pump(const Duration(milliseconds: 7));
+    expect(events.length, 1);
+    expect(events[0].runtimeType, equals(PointerDownEvent));
+    expect(events[0].timeStamp, now() + binding.samplingOffset);
+    expect(events[0].position, Offset(5.0 / ui.window.devicePixelRatio, 0.0));
+
+    // Now the system time is epoch + 9ms
+    await tester.pump(const Duration(milliseconds: 2));
+    expect(events.length, 2);
+    expect(events[1].timeStamp, now() + binding.samplingOffset);
+    expect(events[1].runtimeType, equals(PointerMoveEvent));
+    expect(events[1].position, Offset(25.0 / ui.window.devicePixelRatio, 0.0));
+    expect(events[1].delta, Offset(20.0 / ui.window.devicePixelRatio, 0.0));
+
+    // Now the system time is epoch + 11ms
+    await binding.pump(const Duration(milliseconds: 2));
```

Comment:
Why is this `binding.pump` instead of `tester.bump` as line 97 and 104?

Reply:
This is a nit. I changed it to `tester.pump`